internal/log/log.go

```diff
@@ -8,42 +8,59 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
-func init() {
+type Logger interface {
+	Info(msg string)
+	Debug(msg string)
+	Error(msg string, err error)
+	Fatal(msg string, err error)
+	Warn(msg string)
+	Request(msg string, r *http.Request)
+}
+
+type Log struct{}
+
+func NewLogger() *Log {
 	log.Debug().Msg("Logger started")
+	return &Log{}
 }
 
-func Info(msg string) {
+func (l Log) Info(msg string) {
 	log.Info().Msg(msg)
 }
 
-func Debug(msg string) {
+func (l Log) Debug(msg string) {
 	log.Debug().Msg(msg)
 }
 
-func Error(msg string, err error) {
+func (l Log) Error(msg string, err error) {
 	log.Error().Err(err).Msg(msg)
 }
 
-func Fatal(msg string, err error) {
+func (l Log) Fatal(msg string, err error) {
 	log.Fatal().Err(err).Msg(msg)
 }
 
-func Warn(msg string) {
+func (l Log) Warn(msg string) {
 	log.Warn().Msg(msg)
 }
 
-func Request(msg string, r *http.Request) {
-	defer r.Body.Close()
+func (l Log) Request(msg string, r *http.Request) {
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			panic("error closing http io reader")
+		}
+	}(r.Body)
 	var body interface{}
 	rawbody, err := io.ReadAll(r.Body)
 	if err != nil {
-		Error("Error reading request body", err)
+		l.Error("Error reading request body", err)
```

Comment:
On some of the methods where they are only a single line or two, I'm kind of ok with short variable names. However, as we get into longer functions it makes more sense to use more descriptive names.  If we just want to make things simpler and say "always use descriptive variable names" I'd be fine with it also.